### PR TITLE
Support for Btrieve through PHAPI & Real-Mode Memory Addressing

### LINKED ISSUE #58: [MBMGEMP] Galactic Empire - Crashes on startup
**Module Information**
* Identifier: MBMGEMP
* Module Name: Galactic Empire
* Version: 3.2e

**Describe the bug**
Game crashes on startup of MBBSemu

**To Reproduce**
1.  Attempt to load game

**Expected behavior**
Game loads

**Screenshots**
```
2020-08-24 04:52:40.1280 Warn MBBSEmu.Module.MbbsModule.Execute MBMGEMP exhausted execution Units, creating additional
Critical Exception has occured:
System.FormatException: Input string was not in a correct format.
   at System.Number.ThrowOverflowOrFormatException(ParsingStatus status, TypeCode type)
   at System.Int16.Parse(ReadOnlySpan`1 s, NumberStyles style, IFormatProvider provider)
   at MBBSEmu.Module.McvFile.GetNumeric(Int32 ordinal)
   at MBBSEmu.HostProcess.ExportedModules.Majorbbs.numopt()
   at MBBSEmu.HostProcess.ExportedModules.Majorbbs.Invoke(UInt16 ordinal, Boolean offsetsOnly)
   at MBBSEmu.HostProcess.ExecutionUnits.ExecutionUnit.ExternalFunctionDelegate(UInt16 ordinal, UInt16 functionOrdinal)
   at MBBSEmu.CPU.CpuCore.Tick()
   at MBBSEmu.HostProcess.ExecutionUnits.ExecutionUnit.Execute(IntPtr16 entryPoint, UInt16 channelNumber, Boolean simulateCallFar, Boolean bypassState, Queue`1 initialStackValues, UInt16 initialStackPointer)
   at MBBSEmu.Module.MbbsModule.Execute(IntPtr16 entryPoint, UInt16 channelNumber, Boolean simulateCallFar, Boolean bypassSetState, Queue`1 initialStackValues, UInt16 initialStackPointer)
   at MBBSEmu.HostProcess.MbbsHost.AddModule(MbbsModule module)
   at MBBSEmu.Program.Run(String[] args)
```

**Software Information:**
- OS: Windows 2019 Server Standard
- Telnet Client: NetRunner

### PR DESCRIPTION
**Pull request by enusbaum**
Fixes #58 but Galactic Empire will still have an error on entry of unknown x87 CPU Opcode `FDIVP` (we can log that as a new issue). This PR will get past the PHAPI/Real-Mode access to Btrieve that Galactic Empire does on startup.

Also allocate memory addresses starting at `0x200` as Real-Mode Addresses. Variables start at `0x100` which gives 16MB of memory space for variables before overlapping into the Real-Mode range.

### CODE REVIEW

**Comment by enusbaum** [comment]
One thing I want to call out is that I had to make `BtrievePointerDictionaryNew` static in `ExportedModuleBase.cs` so both `PHAPI` and `MAJORBBS` would be aware of the currently open Btrieve files. I'm not sure if this is the BEST solution... but it works.

File: MBBSEmu/HostProcess/ExportedModules/ExportedModuleBase.cs [hunk 1]
```diff
@@ -34,7 +34,7 @@ public abstract class ExportedModuleBase
         ///     Pointers to files opened using FOPEN
         /// </summary>
         private protected readonly PointerDictionary<FileStream> FilePointerDictionary;
-        private protected readonly Dictionary<IntPtr16, BtrieveFileProcessor> BtrievePointerDictionaryNew;
+        private protected static readonly Dictionary<IntPtr16, BtrieveFileProcessor> BtrievePointerDictionaryNew = new Dictionary<IntPtr16, BtrieveFileProcessor>();
```
Comment (paladine) [inline MBBSEmu/HostProcess/ExportedModules/ExportedModuleBase.cs:37]: this makes me incredibly sad. Can we create a shared dictionary and pass it into the constructor of each ExportedModuleBase subclass?

Comment (enusbaum) [reply MBBSEmu/HostProcess/ExportedModules/ExportedModuleBase.cs:37]: I feel like this is probably the use case for static, when you want to share an object within a base class between implementing members. A private protected static in an abstract class is pretty encapsulated

**Comment by paladine** [comment]
Needs merging, but LGTM

**Review by paladine — APPROVED** [review]

**Pull request merged**